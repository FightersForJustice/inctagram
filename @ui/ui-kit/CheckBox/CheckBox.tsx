import { InputHTMLAttributes, useState } from 'react'
import s from './checkBox.module.scss'

type CheckBoxType = {
  children?: React.ReactNode
  checked?: boolean
  disabled?: boolean
}

export const CheckBox: React.FC<CheckBoxType> = ({ children, checked, disabled }) => {
  return (
    <label className={s.label}>
      <input type="checkbox" className={s.input} defaultChecked={checked} disabled={disabled} />
      {disabled ? (
        <div className={s.blockDisabled}>
          <span className={s.checkbox}></span>
          <span className={s.checkboxOn}>
            <span className={s.check}></span>
          </span>
        </div>
      ) : (
        <div className={s.block}>
          <span className={s.checkbox}>
            <span className={s.checkboxOn}>
              <span className={s.check}></span>
            </span>
          </span>
        </div>
      )}

      <span className={s.text}>{children}</span>
    </label>
  )
}
