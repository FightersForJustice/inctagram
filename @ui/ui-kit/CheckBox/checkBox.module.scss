@use '../../design/settings/colors.scss';

.label {
  position: relative;
  width: 100%;
  .text {
    color: colors.$light-100;
  }
  .input {
    display: none;
  }
  .block {
    width: 36px;
    height: 36px;
    position: relative;
    display: inline-block;
    border-radius: 50%;
    &:hover {
      background-color: colors.$dark-300;
    }
    &:focus {
      background-color: colors.$dark-500;
    }
    &:active {
      background-color: colors.$dark-100;
    }
    .checkbox {
      position: absolute;
      top: 50%;
      left: 50%;
      transform: translate(-50%, -50%);
      width: 18px;
      height: 18px;
      border: 2px solid colors.$light-100;
      border-radius: 3px;
    }
    .checkboxError {
      position: absolute;
      top: 50%;
      left: 50%;
      transform: translate(-50%, -50%);
      width: 18px;
      height: 18px;
      border: 2px solid colors.$danger-500;
      border-radius: 3px;
    }
    .checkboxOn {
      color: colors.$light-100;
      display: block;
      background-color: colors.$light-100;
      opacity: 0;
      transition: 0.2s;
      width: 100%;
      height: 100%;
      border: 2px solid colors.$light-100;
      &:before {
        content: '✔';
        color: colors.$dark-900;
        display: block;
        position: absolute;
        top: -5px;
        left: 1px;
        font-size: 17px;
      }
    }
  }
  .blockDisabled {
    width: 36px;
    height: 36px;
    position: relative;
    display: inline-block;
    cursor: not-allowed;
    .checkbox {
      display: block;
      position: absolute;
      top: 50%;
      left: 50%;
      opacity: 0;
      transition: 0.2s;
      transform: translate(-50%, -50%);
      width: 18px;
      height: 18px;
      border: 2px solid colors.$light-900;
      border-radius: 3px;
    }
    .checkboxOn {
      position: absolute;
      top: 50%;
      left: 50%;
      transform: translate(-50%, -50%);
      width: 18px;
      height: 18px;
      border: 2px solid colors.$dark-100;
      border-radius: 3px;
      display: block;
      background-color: colors.$dark-100;
      opacity: 0;
      transition: 0.2s;
      &:before {
        content: '✔';
        color: colors.$light-100;
        display: block;
        position: absolute;
        top: -4px;
        left: 1px;
      }
    }
  }
  .input:checked + .block .checkbox .checkboxOn {
    opacity: 1;
  }
  .input:checked + .blockDisabled .checkboxOn {
    opacity: 1;
  }
  .input:not(:checked) + .blockDisabled .checkbox {
    opacity: 1;
  }
}
