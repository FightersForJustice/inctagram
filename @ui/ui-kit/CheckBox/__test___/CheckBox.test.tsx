import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { CheckBox } from '../CheckBox'; 

describe('CheckBox component', () => {
  it('renders unchecked checkbox by default', () => {
    const { getByLabelText } = render(<CheckBox>Check me</CheckBox>);
    const checkbox = getByLabelText('Check me');
    expect(checkbox).not.toBeChecked();
  });

  it('renders checked checkbox when "checked" prop is true', () => {
    const { getByLabelText } = render(<CheckBox checked>Check me</CheckBox>);
    const checkbox = getByLabelText('Check me');
    expect(checkbox).toBeChecked();
  });

  it('renders disabled checkbox when "disabled" prop is true', () => {
    const { getByLabelText } = render(<CheckBox disabled>Check me</CheckBox>);
    const checkbox = getByLabelText('Check me');
    expect(checkbox).toBeDisabled();
  });

  it('matches snapshot', () => {
    const { container } = render(<CheckBox>Check me</CheckBox>);
    expect(container).toMatchSnapshot();
  });
});