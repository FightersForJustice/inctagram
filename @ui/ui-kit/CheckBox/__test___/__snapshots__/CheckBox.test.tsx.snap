// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox component matches snapshot 1`] = `
<div>
  <label
    class="label"
  >
    <input
      class="input"
      type="checkbox"
      value=""
    />
    <div
      class="block"
    >
      <span
        class="checkbox"
      >
        <span
          class="checkboxOn"
        >
          <span
            class="check"
          />
        </span>
      </span>
    </div>
    <span
      class="text"
    >
      Check me
    </span>
  </label>
</div>
`;
